### ISSUE TITLE
Investigate and Adopt KEP 3157 Watch List in Istio

### ISSUE THREAD

**Issue by keithmattix** [istio/istio#54344]
KEP is here: https://github.com/kubernetes/enhancements/issues/3157

Looks like we just need to set an env var based on what we detect the kube version to be during install?

**Comment by howardjohn** [comment]
I don't think it needs to be conditional enabled by version, client-go handles that already IIUC

**Comment by hzxuzhonghu** [comment]
Not need a env, just automatically detect kubernetes version?

**Comment by howardjohn** [comment]
Istio shouldn't auto-enable non-stable features so at best we should add
auto-detection once it goes GA
